pkg/config/sampling/sampling.go

```diff
@@ -40,7 +40,10 @@ func (u *Config) Get() *v1.ConfigMap {
 		return nil
 	}
 
-	logrus.WithField("instance", u.jaeger.Name).Debug("Assembling the Sampling configmap")
+	log.WithFields(log.Fields{
```

Comment:
I know that you have turned down my opinions but I find the current logging rather ad hoc. I am pretty suer sooner or later somebody will log something without proper fields.

I like static access to logger but in this case I would rather pass it explicitly or use MDC to for consistent logging and avoid repetition every time you access the logger. Another approach is to have a factory method `creteLogger(j v1alpha1.Jaeger, tuples...) logger` in utils package.

Reply:
> somebody will log something without proper fields

This happens already, as this PR shows: quite a few places had to be fixed to include the instance and/or namespace.

I'm not against using a base logger for the request-scoped code, but I don't quite like the idea of changing method signatures to include a logger which can be derived from the other parameters (`log.WithFields("name": jaeger.Name, "namespace":jaeger.Namespace)`)

A helper like you proposed would certainly be something to consider, though.

Reply:
I added a new commit in the direction of your suggestion. How does it look now?

Reply:
it looks much better :+1: